cpp/src/arrow/csv/parser.cc

```diff
@@ -184,6 +184,31 @@ class SSE42Filter {
   const BulkFilterType filter_;
 };
 
+#elif defined ARROW_HAVE_NEON
+
+// NEON filter: 8 bytes at a time, comparing with all special chars
+
+class NeonFilter {
```

Comment:
Is it doable to rewrite this using `xsimd`? That would make it more easily available with different SIMD instruction sets.

Reply:
The code is using legacy 64 bit SIMD registers (uint8x8_t, __m64), xsimd doesn't support them.
As WordType is 64 bits, using 128 bit register here is inconvenient and hurts performance per my test.
I didn't try changing WordType to 128 bits, guess it may influence other tests.

Reply:
I see. Perhaps you can try changing to 128-bit, but it might decrease performance.

Reply:
See big regression (~30%) for long fields if using 128 bit WordType.